src/ASN1/ASN1Object.php

```diff
@@ -353,10 +369,17 @@ public function findChildrenByType($className)
         return self::fromBinary($file);
     }
 
+    /**
+     * In case we have to provide object as as, but won't provide access to parent node.
+     * That's very important in CMS package, cause we do not wont to give an
+     * ability to change any part of ASN.1 structure.
+     *
+     * @return ASN1ObjectInterface
+     */
     public function detach()
     {
-        $object       = clone $this;
-        $this->parent = null;
+        $object = clone $this;
```

Comment:
It seems to be a mistake, but it would be a BC

Reply:
what is BC?

Reply:
I mean BC break. Someone could use this function and rely on that mistake. I will merge this PR and have to think it would be major or minor version.